mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -67,7 +67,8 @@ def MIOpen_Conv2DBwdWeightOp :
     MIOpen_Op<"conv2d_bwd_weight">,
     Arguments<(ins MemRefRankOf<[F32, F16, BF16], [5]>:$filter,
                    MemRefRankOf<[F32, F16, BF16], [5]>:$input,
-                   MemRefRankOf<[F32, F16, BF16], [5]>:$output)> {
+                   MemRefRankOf<[F32, F16, BF16], [5]>:$output,
+                   Variadic<MemRefRankOf<[F32], [5]>>:$workspace)> {
```

Comment:
To be honest, I almost feel like this should be a different high level op.

Reply:
I don't agree. Adding one extra top-level op would incur additional overheads in tools such as `Conv2dGenerator` and `miopen-gen`.

I will try make this `Variadic` be `Optional` on the other hand.

Reply:
Yeah, I vote for Optional here too